azure/packages/test/scenario-runner/src/ScenarioRunner.ts

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ChildProcess } from "child_process";
+import { TypedEventEmitter, delay } from "@fluidframework/common-utils";
+import {
+	IRunConfig,
+	IRunner,
+	IRunnerEvents,
+	IRunnerStatus,
+	IScenarioConfig,
+	IScenarioRunConfig,
+	RunnerStatus,
+} from "./interface";
+import { convertConfigToScriptParams, createChildProcess } from "./utils";
+
+export abstract class ScenarioRunner<
```

Comment:
Love it! Simplifies life of other runners. They focus on core scenarios they are describing and not mechanics of running tests.